Kernel/FileSystem/Ext2FileSystem.cpp

```diff
@@ -449,11 +451,12 @@ Vector<Ext2FS::BlockIndex> Ext2FS::block_list_for_inode_impl(const ext2_inode& e
     auto process_block_array = [&](unsigned array_block_index, auto&& callback) {
         if (include_block_list_blocks)
             callback(array_block_index);
-        auto array_block = ByteBuffer::create_uninitialized(block_size());
-        read_block(array_block_index, array_block.data());
+        unsigned count = min(blocks_remaining, entries_per_block);
+        size_t read_size = count * sizeof(const __u32*);
```

Comment:
This calculation is wrong, it should use `sizeof(u32)` and not `sizeof(u32*)` :)

Reply:
Good catch!